portal-web/test/functional/com/liferay/portalweb/macros/analyticscloud/ACSegments.macro

```diff
@@ -171,6 +171,18 @@ definition {
 		Click(locator1 = "ACSegments#CANCEL_SEGMENT");
 	}
 
+	macro checkDataSourceOrigin {
+		var key_dataSource = "${dataSource}";
+
+		Click(locator1 = "ACSegments#SELECT_BUTTON");
+
+		ACSegments.searchSegment(findSegment = "${documentTitle}");
+
+		AssertElementPresent(locator1 = "ACSegments#DATA_SOURCE_SEGMENT");
```

Comment:
Hey @calvinctvo, I think the `viewGenericText` macro would work for you in this validation. What do you think?